server/src/test/java/org/elasticsearch/health/HealthServiceTests.java

```diff
@@ -84,11 +132,78 @@ public void testMissingIndicator() {
                 createMockHealthIndicatorService(shardsAvailable)
             )
         );
+        NodeClient client = getTestClient(HealthInfo.EMPTY_HEALTH_INFO);
+        assertGetHealthThrowsException(
+            service,
+            client,
+            "indicator99",
+            false,
+            ResourceNotFoundException.class,
+            "Did not find indicator indicator99",
+            true
+        );
+    }
 
-        expectThrows(ResourceNotFoundException.class, "Did not find indicator indicator99", () -> service.getHealth("indicator99", false));
+    private <T extends Throwable> void assertGetHealthThrowsException(
+        HealthService service,
+        NodeClient client,
+        String indicatorName,
+        boolean explain,
+        Class<T> expectedType,
+        String expectedMessage,
+        boolean expectedInOnFail
```

Comment:
Sorry, I do not fully get what this argument stands for. Can you explain?

Reply:
Ah, the throwable is expected to be passed through the listener. Maybe `expectedViaTheListener` or `expectedOnFailureCalled`? Or something else?